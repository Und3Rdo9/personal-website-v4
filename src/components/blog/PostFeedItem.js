import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const PostFeedItem = ({ post }) => {
  const { title, summary, coverImage, slug } = post;
  return (
    <article className='post-feed__item'>
      <img
        className='post-feed__item-cover'
        src={coverImage.url}
        alt={title}
      />
      <h3 className='post-feed__item-title'>{title}</h3>
      <p className='post-feed__item-summary'>{summary}</p>
      <Link to={`/frontend-blog/${slug}`} className='post-feed__item-cta button button--primary'>Read more</Link>
    </article>
  )
}

PostFeedItem.propTypes = {
  post: PropTypes.shape({
    title: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
    coverImage: PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired,
    dateAndTime: PropTypes.string.isRequired
  }).isRequired
};

export default PostFeedItem;
