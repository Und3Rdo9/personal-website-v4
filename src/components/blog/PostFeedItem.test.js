import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PostFeedItem from './PostFeedItem';

configure({ adapter : new Adapter() });

describe('<PostFeedItem />', () => {

  const post = {
    title: 'testTitle',
    summary: 'testSummary',
    slug: 'test-slug',
    coverImage: {
      url: 'http://timbakkum.net'
    },
    dateAndTime: '21380921389'
  };

  const wrapper = shallow(
    <PostFeedItem post={post} />
  );

  it('renders correctly ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays the correct title', () => {
    expect(wrapper.find('.post-feed__item-title').text()).toBe('testTitle');
  });

  it('displays the correct title', () => {
    expect(wrapper.find('.post-feed__item-summary').text()).toBe('testSummary');
  });

  it('renders the correct link', () => {
    expect(wrapper.find('.post-feed__item-cta').prop('to')).toBe('/frontend-blog/test-slug');
  });

  it('displays the correct image url', () => {
    expect(wrapper.find('.post-feed__item-cover').prop('src')).toEqual('http://timbakkum.net');
  })
});
