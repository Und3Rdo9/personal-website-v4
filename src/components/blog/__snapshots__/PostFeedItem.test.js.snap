// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostFeedItem /> renders correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostFeedItem
    post={
        Object {
            "coverImage": Object {
              "url": "http://timbakkum.net",
            },
            "dateAndTime": "21380921389",
            "slug": "test-slug",
            "summary": "testSummary",
            "title": "testTitle",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="testTitle"
          className="post-feed__item-cover"
          src="http://timbakkum.net"
/>,
        <div
          className="post-feed__item-content"
>
          <h3
                    className="post-feed__item-title"
          >
                    testTitle
          </h3>
          <p
                    className="post-feed__item-summary"
          >
                    testSummary
          </p>
          <Link
                    aria-label="testTitle"
                    className="post-feed__item-cta"
                    replace={false}
                    to="/frontend-blog/test-slug"
          >
                    Read more
          </Link>
</div>,
      ],
      "className": "post-feed__item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "testTitle",
          "className": "post-feed__item-cover",
          "src": "http://timbakkum.net",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="post-feed__item-title"
>
              testTitle
</h3>,
            <p
              className="post-feed__item-summary"
>
              testSummary
</p>,
            <Link
              aria-label="testTitle"
              className="post-feed__item-cta"
              replace={false}
              to="/frontend-blog/test-slug"
>
              Read more
</Link>,
          ],
          "className": "post-feed__item-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testTitle",
              "className": "post-feed__item-title",
            },
            "ref": null,
            "rendered": "testTitle",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testSummary",
              "className": "post-feed__item-summary",
            },
            "ref": null,
            "rendered": "testSummary",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "testTitle",
              "children": "Read more",
              "className": "post-feed__item-cta",
              "replace": false,
              "to": "/frontend-blog/test-slug",
            },
            "ref": null,
            "rendered": "Read more",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="testTitle"
            className="post-feed__item-cover"
            src="http://timbakkum.net"
/>,
          <div
            className="post-feed__item-content"
>
            <h3
                        className="post-feed__item-title"
            >
                        testTitle
            </h3>
            <p
                        className="post-feed__item-summary"
            >
                        testSummary
            </p>
            <Link
                        aria-label="testTitle"
                        className="post-feed__item-cta"
                        replace={false}
                        to="/frontend-blog/test-slug"
            >
                        Read more
            </Link>
</div>,
        ],
        "className": "post-feed__item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "testTitle",
            "className": "post-feed__item-cover",
            "src": "http://timbakkum.net",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="post-feed__item-title"
>
                testTitle
</h3>,
              <p
                className="post-feed__item-summary"
>
                testSummary
</p>,
              <Link
                aria-label="testTitle"
                className="post-feed__item-cta"
                replace={false}
                to="/frontend-blog/test-slug"
>
                Read more
</Link>,
            ],
            "className": "post-feed__item-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testTitle",
                "className": "post-feed__item-title",
              },
              "ref": null,
              "rendered": "testTitle",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testSummary",
                "className": "post-feed__item-summary",
              },
              "ref": null,
              "rendered": "testSummary",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "testTitle",
                "children": "Read more",
                "className": "post-feed__item-cta",
                "replace": false,
                "to": "/frontend-blog/test-slug",
              },
              "ref": null,
              "rendered": "Read more",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
